src/index.coffee

```diff
@@ -42,12 +42,21 @@ class ConnectApp
     @app.use connect.directory(if typeof @root == "object" then @root[0] else @root)
 
     if @https
-      @server = (http2 || https).createServer
-        key: @https.key || fs.readFileSync __dirname + '/certs/server.key'
-        cert: @https.cert || fs.readFileSync __dirname + '/certs/server.crt'
-        ca: @https.ca || fs.readFileSync __dirname + '/certs/ca.crt'
-        passphrase: @https.passphrase || 'gulp'
-        @app
+
+      # use some defaults when not set. do not touch when a key is already specified
+      # see https://github.com/AveVlad/gulp-connect/issues/172
+      if typeof (@https) is 'boolean' || !@https.key
+
+        # change it into an object if it is not already one
+        if !(typeof (@https) is "object")
+          @https = {}
+
+        @https.key        = fs.readFileSync __dirname + '/certs/server.key'
+        @https.cert       = fs.readFileSync __dirname + '/certs/server.crt'
+        @https.ca         = fs.readFileSync __dirname + '/certs/server.crt'
```

Comment:
Apparently openssl sucks on windows, I will have to wait till I get to work tomorrow.